src/_includes/components/macros.njk

```diff
@@ -14,14 +14,8 @@
     </p>
   </div>
 </div>
-{% endmacro %}
+{%- endmacro -%}
 
-{% macro link(key, text) %}
-  {% set url = "" %}
-    {% for k, v in glossary %}
```

Comment:
You don't have access to the data variables in a nunjucks macro, especially ones added through yaml. If you want something like this, we'd have better luck if it was JS/JSON.